Makefile

```diff
@@ -30,15 +34,32 @@ EMPTY :=
 SPACE := $(EMPTY) $(EMPTY)
 JAVA_CLASS_PATH := $(subst $(SPACE),:,$(JARS_TO_DOWNLOAD))
 
+CONSUMER ?= consumer
+
 $(JARS_TO_DOWNLOAD):
 	mkdir -p `dirname $@`
 	curl -s -L -o $@ -O $(URL_PREFIX)`echo $@ | sed 's/$(JAR_DIR)\///g'`
 
 download_jars: $(JARS_TO_DOWNLOAD)
 
 build:
-	CGO_ENABLED=0 go build -installsuffix cgo -o build/consumer $(PKG)/cmd/consumer
+	CGO_ENABLED=0 go build -installsuffix cgo -o build/$(CONSUMER) $(PKG)/cmd/$(CONSUMER)
 
 run: build download_jars
 	command -v java >/dev/null 2>&1 || { echo >&2 "Java not installed. Install java!"; exit 1; }
-	java -cp $(JAVA_CLASS_PATH) com.amazonaws.services.kinesis.multilang.MultiLangDaemon consumer.properties
+	java -cp $(JAVA_CLASS_PATH) \
+	com.amazonaws.services.kinesis.multilang.MultiLangDaemon \
+	$(CONSUMER).properties
+
+bench:
+	go test -bench=. github.com/Clever/amazon-kinesis-client-go/decode/
+
+test: $(PKGS)
+$(PKGS): golang-test-all-deps
+	$(call golang-test-all,$@)
+
+$(GOPATH)/bin/glide:
+	@go get github.com/Masterminds/glide
+
+install_deps: $(GOPATH)/bin/glide
+	@$(GOPATH)/bin/glide install
```

Comment:
We also should someday update the `Dockerfile` to run the java command directly instead of via `make`, to enable passing in signals / graceful shutdown.